atlasdb-config/src/main/java/com/palantir/atlasdb/factory/PersistentStorages.java

```diff
@@ -17,20 +17,43 @@
 package com.palantir.atlasdb.factory;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import com.google.common.base.MoreObjects;
 import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
 
-public final class PersistentStorageFactories {
+public final class PersistentStorages {
     private static final Pattern UUID_PATTERN = Pattern.compile(
             "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
 
     private static final Set<String> SANITIZED_PATHS = new HashSet<>();
 
-    private PersistentStorageFactories() {}
+    private PersistentStorages() {}
+
+    /**
+     * Given the {@code absolutePath} deletes all sub-folders, sub-files and the folder pointed by it.
+     *
+     * @param absolutePath which we want to delete
+     * @throws IOException if there is an underlying exception
+     */
+    public static void deletePath(Path absolutePath) throws IOException {
+        try (Stream<Path> stream = Files.walk(absolutePath)) {
```

Comment:
Some notes:
- should we validate that the path is an absolute path?
- it feels this method should be private, unless I'm missing something.